lib/usd/pxrUsdPreviewSurface/usdPreviewSurfaceWriter.cpp

```diff
@@ -45,11 +46,18 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
-
-PXRUSDMAYA_REGISTER_WRITER(
+PXRUSDMAYA_REGISTER_SHADER_WRITER(
     pxrUsdPreviewSurface,
     PxrMayaUsdPreviewSurface_Writer);
 
+UsdMayaShaderWriter::ContextSupport
+PxrMayaUsdPreviewSurface_Writer::CanExport(const UsdMayaJobExportArgs& exportArgs)
+{
+    return exportArgs.renderContext == UsdMayaShadingModeTokens->preview
+        ? ContextSupport::Supported
+        : ContextSupport::Unsupported;
```

Comment:
Shouldn't this at least be fallback? If your renderContext is something other than the universal context, and no other writer offers support, would you get no shader authored for a pxrUsdPreviewSurface node in Maya?